src/corona_hakab_model/consts.py

```diff
@@ -102,9 +71,42 @@
         lambda agent: agent.medical_state.name == "Recovered",
         lambda agent: agent.medical_state.name == "Symptomatic",
     ),  # TODO: Define better API
-    # dictionary of {date : percent} that controls what percentage of schools are open
-    "school_openage_factors": {1: 0, 30: 0.5, 45: 1},
-    "should_change_school_openage": False
+
+    # policies data
+    # a dictionary of type day:List[ConnectionTypes]. on each day, keeps only the given connection types opened
+    "policies_changes": {
+        40: ([ConnectionTypes.Family, ConnectionTypes.Other], "closing schools and works"),
+        70: ([ConnectionTypes.Family, ConnectionTypes.Other, ConnectionTypes.School], "opening schools"),
+        100: (ConnectionTypes, "opening works")
+    },
+    "change_policies": False,
+
+    # policies acting on a specific connection type, when a term is satisfied
+    "partial_opening_active": True,
+    # each connection type gets a list of conditioned policies
+    # each conditioned policy actives a specific policy when a condition is satisfied
+    # each policy changes the multiplication factor of a specific circle
+    # each policy is activated only if a list of terms is full-filled.
+    "connection_type_to_conditioned_policy": {
+        ConnectionTypes.School:
+        [
+            ConditionedPolicy(
+                activating_condition=lambda kwargs: len(np.flatnonzero(kwargs["manager"].contagiousness_vector)) > 1000,
+                policy=Policy(0, [lambda circle: random() > 0]), message="closing all schools"),
```

Comment:
a. random() > 0 always returns true
b. calling random() for each circle is slow

Reply:
i know, this is only there as a prove of concept that you can close a given percent of the circles at a given connection type (this was the requirement of integration team).
the random() call only occurs if the first condition is true, and the policy is not already active, making it occur only on rare occasions, so it is ok time-wise to roll for every circle.